checkbox-support/checkbox_support/dbus/gnome_monitor.py

```diff
@@ -104,6 +106,107 @@ def set_extended_mode(self) -> Dict[str, str]:
         self._apply_monitors_config(state[0], extended_logical_monitors)
         return configuration
 
+    def cycler(
+        self,
+        res: bool = True,
+        max_res_amount: int = 5,
+        trans: bool = False,
+        log: Callable[..., Any] = None,
+        action: Callable[..., Any] = None,
+        **kwargs
+    ):
+        """
+        Cycling change the monitors configurations automatically
+
+        Args:
+            res: Cycling the resolution or not.
+
+            max_res_amount: The supported resolution might be a lot
+                            and this could limited test amount.
+
+            trans: Cycling the transform or not.
+
+            log: For logging, the string is constructed by
+                 [monitor name]_[resolution]_[transform]_.
+
+            action: For extra steps for each cycle,
+                    the string is constructed by
+                    [monitor name]_[resolution]_[transform]_.
+                    Please note that the delay is needed inside this
+                    callback to wait the monitors to response
+        """
+        monitors = []
+        modes_list = []
+        # ["normal": 0, "left": 1, "inverted": 6, "right": 3]
+        trans_list = [0, 1, 6, 3] if trans else [0]
+
+        # for multiple monitors, we need to create res combination
+        state = self._get_current_state()
+        for monitor, modes in state[1].items():
+            monitors.append(monitor)
+            modes_list.append(self._resolution_filter(modes, max_res_amount))
+        mode_combination = list(itertools.product(*modes_list))
+
+        for mode in mode_combination:
+            for trans in trans_list:
+                logical_monitors = []
+                position_x = 0
+                m_list = list(mode)
+                uni_string = ""
+                for monitor in monitors:
```

Comment:
I hope I got this right. Instead of creating `m_list` and iterating through it via `.pop(0)` why don't we instead use something like:
```
for monitor, m in zip(monitors, mode):
```